### PR TITLE
Bump to Rust 1.34.0

### PR DESCRIPTION
**Pull request by christophermaier**

### CODE REVIEW

File: components/core/src/crypto/keys/box_key_pair.rs [hunk 1]
```diff
@@ -497,13 +497,12 @@ mod test {
         assert_eq!(pair.public, None);
         match pair.public() {
             Ok(_) => panic!("Empty pair should not have a public key"),
-            Err(Error::CryptoError(_)) => assert!(true),
+            Err(Error::CryptoError(_)) => { /* OK */ }
```
Comment (baumanj) [inline components/core/src/crypto/keys/box_key_pair.rs]: We could simplify this a bit:
```rust
if let Err(Error::CryptoError(_)) = pair.public() {
    // OK
} else {
    panic!("Expected Error::CryptoError");
}
```
It reduces the information that's produced in the case of failing with a _different_ error, but I'm ok with that tradeoff. I'll leave it up to you.

File: components/core/src/crypto/keys/box_key_pair.rs [hunk 2]
```diff
@@ -497,13 +497,12 @@ mod test {
...
             e => panic!("Unexpected error: {:?}", e),
         }
         assert_eq!(pair.secret, None);
-        match pair.secret() {
-            Ok(_) => panic!("Empty pair should not have a secret key"),
-            Err(_) => assert!(true),
+        if pair.secret().is_ok() {
+            panic!("Empty pair should not have a secret key");
```
Comment (baumanj) [inline components/core/src/crypto/keys/box_key_pair.rs]: Or
```rust
assert!(pair.secret().is_err(), "Empty pair should not have a secret key");
```
Similarly elsewhere

File: components/core/src/package/archive.rs [hunk 3]
```diff
@@ -411,7 +410,7 @@ impl PackageArchive {
 
         match self.read_metadata(file) {
             Ok(Some(body)) => {
-                let ids: Vec<String> = body.lines().map(|d| d.to_string()).collect();
+                let ids: Vec<String> = body.lines().map(ToString::to_string).collect();
```
Comment (baumanj) [inline components/core/src/package/archive.rs]: I think
```suggestion
                let ids: Vec<String> = body.lines().map(str::to_string).collect();
```
or
```suggestion
                let ids: Vec<String> = body.lines().map(str::to_owned).collect();
```
are more illustrative. Unfortunately clippy makes the suggestion which is most broadly applicable, but when our type is know beyond just the trait bound, I find it makes it clearer what's going on.

File: components/core/src/package/ident.rs [hunk 4]
```diff
@@ -528,9 +526,8 @@ mod tests {
                                   "banana".to_string(),
                                   Some("1.0.0".to_string()),
                                   Some("20150521131555".to_string()));
-        match a.partial_cmp(&b) {
-            Some(_) => panic!("We tried to return an order"),
-            None => assert!(true),
+        if a.partial_cmp(&b).is_some() {
+            panic!("We tried to return an order");
```
Comment (baumanj) [inline components/core/src/package/ident.rs]: Why not
```rust
        assert!(a.partial_cmp(&b).is_none(), "We tried to return an order");
```
?

File: components/core/src/package/metadata.rs [hunk 5]
```diff
@@ -62,7 +61,7 @@ impl FromStr for Bind {
         };
         let exports = match parts.next() {
             None => return Err(Error::MetaFileBadBind),
-            Some(exports) => exports.split(' ').map(|t| t.to_string()).collect(),
+            Some(exports) => exports.split(' ').map(ToString::to_string).collect(),
```
Comment (baumanj) [inline components/core/src/package/metadata.rs]:
```suggestion
            Some(exports) => exports.split(' ').map(str::to_string).collect(),
```

**Review by baumanj — APPROVED** [review]
Some little suggestions, but all optional

File: components/core/src/crypto/keys/box_key_pair.rs (hunk 2 above)
Comment (christophermaier) [reply components/core/src/crypto/keys/box_key_pair.rs]: Yup, that's much better 😄

**Review by raskchanky — APPROVED** [review]

**Pull request merged**